cluster-agent/controllers/argoproj.io/application_controller_test.go

```diff
@@ -566,6 +575,229 @@ var _ = Describe("Application Controller", func() {
 			Expect(reconciledStateObj.Source.TargetRevision).To(Equal(guestbookApp.Status.Sync.ComparedTo.Source.TargetRevision))
 			Expect(reconciledStateObj.Destination.Namespace).To(Equal(guestbookApp.Status.Sync.ComparedTo.Destination.Namespace))
 		})
+
+		It("Test to check whether syncError field is stored in database", func() {
+			By("Close database connection")
+			defer dbQueries.CloseDatabase()
+			defer testTeardown()
+			Expect(err).To(BeNil())
+
+			ctx = context.Background()
+
+			app := appv1.ApplicationCondition{
+				Type:               "SyncError",
+				Message:            "Failed to sync",
+				LastTransitionTime: &metav1.Time{Time: time.Now()},
+			}
+			guestbookApp.Status.Conditions = append(guestbookApp.Status.Conditions, app)
+
+			By("Add databaseID label to applicationID")
+			databaseID := guestbookApp.Labels[dbID]
+			applicationDB := &db.Application{
+				Application_id:          databaseID,
+				Name:                    name,
+				Spec_field:              "{}",
+				Engine_instance_inst_id: gitopsEngineInstance.Gitopsengineinstance_id,
+				Managed_environment_id:  managedEnvironment.Managedenvironment_id,
+			}
+
+			applicationStateDB := &db.ApplicationState{
+				Applicationstate_application_id: applicationDB.Application_id,
+			}
+
+			By("Create Application in Database")
+			err = reconciler.DB.CreateApplication(ctx, applicationDB)
+			Expect(err).To(BeNil())
+
+			By("Verify SyncError is not present in database")
+			err = reconciler.DB.GetApplicationStateById(ctx, applicationStateDB)
+			Expect(err).ToNot(BeNil())
+			Expect(db.IsResultNotFoundError(err)).To(BeTrue())
+
+			By("Create a new ArgoCD Application")
+			err = reconciler.Create(ctx, guestbookApp)
+			Expect(err).To(BeNil())
+
+			By("Call reconcile function")
+			result, err := reconciler.Reconcile(ctx, newRequest(namespace, name))
+			Expect(err).To(BeNil())
+			Expect(result).ToNot(BeNil())
+
+			err = reconciler.DB.GetApplicationStateById(ctx, applicationStateDB)
+			Expect(applicationStateDB.ReconciledState).ToNot(BeNil())
+			Expect(err).To(BeNil())
+
+			// Checking for new values in database object
+			for _, syncError := range guestbookApp.Status.Conditions {
+				Expect(applicationStateDB.SyncError).To(Equal(syncError.Message))
+				Expect(syncError.Type).To(Equal("SyncError"))
```

Comment:
I think this for loop can just be replaced with `Expect(applicationStateDB.SyncError).To(Equal("Failed to sync"))`